Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MediaUnit.java

```diff
@@ -0,0 +1,207 @@
+import org.kitodo.api.dataformat.mets.FileXmlElementAccessInterface;
+import org.kitodo.api.dataformat.mets.UseXmlAttributeAccessInterface;
+import org.kitodo.dataformat.metskitodo.DivType;
+import org.kitodo.dataformat.metskitodo.DivType.Fptr;
+import org.kitodo.dataformat.metskitodo.FileType;
+import org.kitodo.dataformat.metskitodo.Mets;
+
+public class MediaUnit implements FileXmlElementAccessInterface {
+    /**
+     * Each media unit can be available in different variants, for each of which
+     * a media file is available. This is in this map.
+     */
+    private Map<MediaVariant, MediaFile> mediaFiles = new HashMap<>();
+
+    /**
+     * Sequence number of the media unit. The playback order of the media units
+     * when referenced from a structure is determined by this attribute (not by
+     * the order of the references).
+     */
+    private int order;
+
+    /**
+     * A human readable label for the order of this media unit. This need not be
+     * directly related to the order number. Examples of order labels could be
+     * “I, II, III, IV, V,  - , 1, 2, 3”, meanwhile the order would be “1, 2, 3,
+     * 4, 5, 6, 7, 8, 9”.
+     */
+    private String orderlabel;
+
+    /**
+     * Public constructor for a new media unit. This constructor can be used
+     * with the service loader to get a new instance of media unit.
+     */
+    public MediaUnit() {
+    }
+
+    /**
+     * Constructor for developing a media unit from a METS {@code <div>}
+     * element.
+     * 
+     * @param div
+     *            METS {@code <div>} element to be evaluated
+     * @param mets
+     *            the Mets structure is searched for corresponding uses
+     * @param mediaVariants
+     *            list of media variants from which the media variant for the
+     *            given use is taken
+     */
+    MediaUnit(DivType div, Mets mets, Map<String, MediaVariant> mediaVariants) {
+        mediaFiles = div.getFptr().parallelStream().map(Fptr::getFILEID).filter(FileType.class::isInstance)
+                .map(FileType.class::cast)
+                .collect(Collectors.toMap(
+                    file -> mediaVariants.get(mets.getFileSec().getFileGrp().parallelStream()
+                            .filter(fileGrp -> fileGrp.getFile().contains(file)).findFirst()
+                            .orElseThrow(() -> new IllegalArgumentException(
+                                    "Corrupt file: <mets:fptr> not referenced in <mets:fileGrp>"))
+                            .getUSE()),
+                    file -> new MediaFile(file.getFLocat().get(0))));
+        order = div.getORDER().intValue();
+        orderlabel = div.getORDERLABEL();
+    }
+
+    /**
+     * Returns the list of available media variants with the corresponding media
+     * files.
+     * 
+     * @return available media variants with corresponding media files
+     */
+    @Override
+    public Set<Entry<MediaVariant, MediaFile>> getAllUsesWithFLocats() {
```

Comment:
why are you not returning the map?

Reply:
We had already discussed that when we defined the interface: The answer was that otherwise you could manipulate the map arbitrarily. So you get the result of the function `entrySet()` over which you can iterate if you want to process the whole map. But you can not damage it.